HeartBeat.php

```diff
@@ -60,13 +60,8 @@ public function beforeSurveyPage()
     public function newDirectRequest()
     {
         $oEvent = $this->event;
-        $sAction=$oEvent->get('function');
-        if ($oEvent->get('target') == "HeartBeat")
-        {
-            if($sAction == 'beat')
-                $this->actionBeat();
-            else
-                throw new CHttpException(404,'Unknow action');
+        if ($oEvent->get('target') == "HeartBeat") {
+            $this->actionBeat();
```

Comment:
We don't need the $sAction anymore ?

Reply:
No, since 2.5X : you can call directrequest without action.

And here : we have only one action, then why ask one ?